packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -1178,29 +1180,41 @@ class FlutterPlugin implements Plugin<Project> {
         } // end def addFlutterDeps
 
         if (isFlutterAppProject()) {
+
             project.android.applicationVariants.all { variant ->
                 Task assembleTask = getAssembleTask(variant)
                 if (!shouldConfigureFlutterTask(assembleTask)) {
                   return
                 }
                 Task copyFlutterAssetsTask = addFlutterDeps(variant)
                 copyFlutterAssetsTask.doLast {
+
+                  def content = "";
+                  def outputDir = copyFlutterAssetsTask.destinationDir
+                  def publicKeyFile = new File("${project.rootDir.absolutePath}/../public.pem");
+
+                  def shorebirdYamlFile = new File("${outputDir}/flutter_assets/shorebird.yaml")
+                  if (publicKeyFile.exists()) {
+                    def base64EncodedKey = Base64.getEncoder().encode(publicKeyFile.readBytes());
+                    content += 'patch_public_key: ' + new String(base64EncodedKey) + '\n';
+                  }
+
```

Comment:
Is it possible for us to start to move our modifications here out into their own function?  I don't know much about gradle/groovy to know how practical that is (or if it's worth doing).